### ISSUE TITLE
Symbol not found: _clock_gettime when using kqueue native transport on MacOS 10.11.6

### ISSUE THREAD

**Issue by marcuslinke** [netty/netty#6837]
Tried  to use netty-transport-native-kqueue on MacOS 10.11.6 and got the following:

```
dyld: lazy symbol binding failed: Symbol not found: _clock_gettime
  Referenced from: /private/var/folders/y3/8mx9s87n01942z74s5wjsxjw0000hb/T/libnetty-transport-native-kqueue8841250350785055197.dylib (which was built for Mac OS X 10.12)
  Expected in: /usr/lib/libSystem.B.dylib
```

### Netty version
4.1.12.Final

### JVM version (e.g. `java -version`)
java version "1.7.0_80"

### OS version (e.g. `uname -a`)
MacOS 10.11.6 (El Capitan)

**Comment by normanmaurer** [comment]
Yes the transport only works on MacOS Sierra+

**Comment by marcuslinke** [comment]
@normanmaurer Is there any chance to get this compatible with El Capitan?

**Comment by marcuslinke** [comment]
I'm not a C developer but maybe this could be fixed by a simmilar workaround as described in https://github.com/pocoproject/poco/issues/1453?

**Comment by marcuslinke** [comment]
Also found this: https://github.com/jordansissel/xdotool/commit/9bc31403b95826d9fd8d6540c459ee862b2ac127

@normanmaurer @Scottmitch Would it be an accepted workaraound?

**Comment by normanmaurer** [comment]
@marcuslinke I will check next week when I have a few cycles. Stay tuned

**Comment by Scottmitch** [comment]
Note that this is a duplicate of https://github.com/Scottmitch/netty/pull/1#issuecomment-290704233 and there is additional context explained there.

TLDR - transport-native-kqueue should compile against 10.11 but the artifacts we publish are built on 10.12 and we do not support linking to older macOS versions.

**Comment by stevenschlansker** [comment]
[ This comment was originally on the other issue, but as this is the actual project, I'll comment here ]

This is quite frustrating -- a developer brought in testcontainers, which brings in netty, which exposes this issue and he broke the build on some of our developers' machines. We're now forced to choose between upgrading our hardware or using Netty...

I will echo the request that publishing an artifact to Maven Central which does not work on ~ 1/3 the Mac install base (roughly guessing based on the link above) is not very nice, and "just build your own" isn't really a solution when you bring in a third party project that depends on Netty.

**Comment by stevenschlansker** [comment]
Ah, in case anyone else gets frustrated by this -- the problem is that the `docker-java` library is inappropriately pulling in this kqueue transport, when it really doesn't need to.  So I'll work with the library maintainers to fix that...
